### ISSUE TITLE
How to increase numerical accuracy of Pytorch model?

### ISSUE THREAD

**Issue by gentlezr** [pytorch/pytorch#27033]
I write this sentence in my script

`print(self.netG(self.real_A)-self.netG(self.real_A))
`
I think I can get a all zero tensor but no.

```
tensor([[ [[-0.0032,  0.0089, -0.0085,  ..., -0.0027,  0.0004, -0.0022],
          [-0.0019, -0.0022,  0.0775,  ...,  0.0236, -0.0277, -0.0125],
          [ 0.0049,  0.0159,  0.0203,  ..., -0.0212,  0.0010, -0.0069],
          ...,
          [ 0.0042,  0.0081, -0.0127,  ..., -0.0097,  0.0136, -0.0002],
          [-0.0010,  0.0020, -0.0066,  ...,  0.0260,  0.0433,  0.0088],
          [-0.0023,  0.0095,  0.0125,  ...,  0.0005,  0.0090,  0.0029]]]],
       device='cuda:0', grad_fn=<SubBackward0>)
```

**Comment by ailzhang** [comment]
Hi @gentlezr try make them double tensors using `.double()`. Alternatively you can set default dtype `torch.set_default_dtype(torch.float64)` globally. 
Feel free to reopen if this doesn't answer your question. Thanks!

**Comment by gentlezr** [comment]
model.eval()
 This has any effect only on certain modules. See documentations of
        particular modules for details of their behaviors in training/evaluation
        mode, if they are affected, e.g. :class:`Dropout`, :class:`BatchNorm`,
        etc.

**Comment by gentlezr** [comment]
> Hi @gentlezr try make them double tensors using `.double()`. Alternatively you can set default dtype `torch.set_default_dtype(torch.float64)` globally.
> Feel free to reopen if this doesn't answer your question. Thanks!

I find I use dropout in my G model, model.eval() can  solve it. Thanks very much.
